lib/go/go.mod

```diff
@@ -0,0 +1,8 @@
+module github.com/container-storage-interface/spec/lib/go/csi
```

Comment:
maybe this go.mod file should be moved into `lib/go/csi` instead of keeping it adjacent to the Makefile?

Reply:
hmm, it seems logical to have it here since top level csi package declaration exists in this directory.

Reply:
it looks like the (empty) top level file was put there to satisfy go-dep, which didn't support multiple modules per repo unlike `go mod`. i'd rather be more precise with our module definitions.

would strongly prefer to move the go.mod file as suggested.

Reply:
ack.

Reply:
actually, this is tricker. I am okay with moving go.mod file to csi directory but it looks like even though our build is green, having go.mod file in a subdirectory prevents the csi module from being usable. 

afaict - if we want to use go modules, we will have to place go.mod at the root of the repo.